Peering/azure/docs (#475)
kyma-project/cloud-manager: docs/user/tutorials/01-30-azure-vpc-peering.md

# Use VPC peering in Azure

> This tutorial explains how to create VPC peering connection between a remote VPC network and Kyma in Azure. Tutorial 

Reviewer: ```suggestion
This tutorial explains how to create a VPC peering connection between a remote VPC network and Kyma in Azure. The tutorial 
```

> assumes that Cloud Manager module is enabled in your Kyma. Steps from this tutorial create new resource group, VPC 

Reviewer: ```suggestion
assumes that the Cloud Manager module is enabled in your Kyma cluster. Follow the steps from this tutorial to create a new resource group, VPC 
```

> network and VM, and assigns required roles to provided Kyma service principal in your Azure subscription. If you wish to

Reviewer: ```suggestion
network and VM, and assign required roles to the provided Kyma service principal in your Azure subscription. If you want to
```

> use existing resources instead of creating new ones, adjust variable names accordingly and skip

Reviewer: ```suggestion
use the existing resources instead of creating new ones, adjust variable names accordingly and skip
```

the steps that create those resources.

## Steps <!-- {docsify-ignore} -->

> 1. Login to Azure and set active subscription

Reviewer: ```suggestion
1. Login to Azure and set the active subscription
```

> 1. Login to Azure and set the active subscription

Reviewer: ```suggestion
1. Log in to Azure and set the active subscription
```

   ```shell
   export SUBSCRIPTION=<Name or ID of subscription>
   az login
   az account set --subscription $SUBSCRIPTION
   ```

> 2. Assign required roles to Cloud Manager peering service principal

Reviewer: ```suggestion
2. Assign required roles to the Cloud Manager peering service principal
```

   ```shell
   export SUBSCRIPTION_ID=$(az account show --query id -o tsv)
   export PRINCIPAL_NAME=<Cloud Manager service principal name>
   export OBJECT_ID=$(az ad sp list --display-name $PRINCIPAL_NAME --query "[].id" -o tsv)
   
   az role assignment create --assignee $OBJECT_ID \
   --role "Network Contributor" \
   --scope "/subscriptions/$SUBSCRIPTION_ID"
   
   az role assignment create --assignee $OBJECT_ID \
   --role "Classic Network Contributor" \
   --scope "/subscriptions/$SUBSCRIPTION_ID"
   ```
   
3. Set the region that is closest to your Kyma cluster. Use `az account list-locations` to list available locations. 
   ```shell
      export REGION=<Location>
   ```
4. Create a resource group that will be a container for related resources
   ```shell
   export RANDOM_ID="$(openssl rand -hex 3)"
   export RESOURCE_GROUP_NAME="myResourceGroup$RANDOM_ID"
   az group create --name $RESOURCE_GROUP_NAME --location $REGION
   ```

> 5. Create network

Reviewer: ```suggestion
5. Create a network
```

   ```shell
   export VNET_NAME="myVnet$RANDOM_ID"
   export ADDRESS_PREFIX=172.0.0.0/16
   export SUBNET_PREFIX=172.0.0.0/24
   export SUBNET_NAME="MySubnet"
   
   az network vnet create -g $RESOURCE_GROUP_NAME -n $VNET_NAME --address-prefix $ADDRESS_PREFIX --subnet-name $SUBNET_NAME --subnet-prefixes $SUBNET_PREFIX
   ```

> 6. Create virtual machine

Reviewer: ```suggestion
6. Create a virtual machine
```

   64 bytes from 172.0.0.4: icmp_seq=7 ttl=63 time=2.01 ms
   64 bytes from 172.0.0.4: icmp_seq=8 ttl=63 time=4.26 ms
   64 bytes from 172.0.0.4: icmp_seq=9 ttl=63 time=1.89 ms
   64 bytes from 172.0.0.4: icmp_seq=10 ttl=63 time=2.08 ms
   64 bytes from 172.0.0.4: icmp_seq=11 ttl=63 time=2.01 ms
   64 bytes from 172.0.0.4: icmp_seq=12 ttl=63 time=2.24 ms
   64 bytes from 172.0.0.4: icmp_seq=13 ttl=63 time=1.80 ms
   64 bytes from 172.0.0.4: icmp_seq=14 ttl=63 time=4.32 ms
   64 bytes from 172.0.0.4: icmp_seq=15 ttl=63 time=2.03 ms
   64 bytes from 172.0.0.4: icmp_seq=16 ttl=63 time=2.03 ms
   64 bytes from 172.0.0.4: icmp_seq=17 ttl=63 time=5.19 ms
   64 bytes from 172.0.0.4: icmp_seq=18 ttl=63 time=1.86 ms
   64 bytes from 172.0.0.4: icmp_seq=19 ttl=63 time=1.92 ms
   64 bytes from 172.0.0.4: icmp_seq=20 ttl=63 time=1.92 ms
   
   === 172.0.0.4 ping statistics ===
   20 packets transmitted, 20 received, 0% packet loss, time 19024ms
   rtt min/avg/max/mdev = 1.800/3.060/8.096/1.847 ms
   ...
   ```

13. Clean up Kubernetes resources

    * Remove the created workloads:
      ```shell
      kubectl delete -n $NAMESPACE deployment azurevpcpeering-demo
      ```

>     * Remove the created azurevpcpeering:

Reviewer: ```suggestion
    * Remove the created `azurevpcpeering`:
```
